### PR TITLE
Heap refinement Part 4 -- zones merged

### PR DESCRIPTION
**Pull request by keyonjie**
Merge zones SYSTEM_SHARED and RUNTIME_SHARED to the BUFFER zone, and change the corresponding callers.

### CODE REVIEW

File: src/lib/alloc.c [hunk 1]
```diff
@@ -945,7 +942,8 @@ static void *_balloc_unlocked(uint32_t flags, uint32_t caps, size_t bytes,
 		/* Continue from the next heap */
 	}
 
-	return ptr;
+	return (flags & SOF_MEM_FLAG_COHERENT) && (CONFIG_CORE_COUNT > 1) ?
```
Comment (RanderWang) [inline src/lib/alloc.c:890]: if SOF_MEM_FLAG_COHERENT == 1, then we prefer uncache addr ?  the flag name is confusing. Or named SOF_MEM_FLAG_UNCACHE

File: Kconfig.xtos-dbg [hunk 2]
```diff
@@ -12,3 +12,12 @@ config DEBUG_BLOCK_FREE
 	  already freed block of memory. Enabling this feature increases
 	  number of memory writes and reads, due to checks for memory patterns
 	  that may be performed on allocation and deallocation.
+
+config DEBUG_FORCE_COHERENT_BUFFER
+	bool "Force the allocator to allocate coherent buffer only"
```
Comment (RanderWang) [inline Kconfig.xtos-dbg:17]: means uncached buffer ?  I know cache coherence but what is coherence buffer ?

Comment (lgirdwood) [reply Kconfig.xtos-dbg:17]: Yes, so coherent (a generic term) means uncache (coherent) memory on Intel HW.

**Comment by gkbldcig** [comment]
Can one of the admins verify this patch?

**Comment by keyonjie** [comment]
@lgirdwood the CI result is becoming better now.
only fuzzy IPC and test bench issues looks valid to me, will fix them soon.

**Comment by keyonjie** [comment]
> ```c
> CAPS_CACHE
> ```

Hi @lgirdwood let me try to clarify it.

Now the CAPS_CACHE is somewhat confusion, the flag doesn't mean anything, it is listed in all zones in memory.c.

And the usage of it from the topology don't tell any difference wrt this flag:
```
dnl Memory capabilities for different buffer types on Tigerlake
define(`PLATFORM_DAI_MEM_CAP',
        MEMCAPS(MEM_CAP_RAM, MEM_CAP_DMA, MEM_CAP_CACHE, MEM_CAP_HP))
define(`PLATFORM_HOST_MEM_CAP',
        MEMCAPS(MEM_CAP_RAM, MEM_CAP_DMA, MEM_CAP_CACHE, MEM_CAP_HP))
define(`PLATFORM_PASS_MEM_CAP',
        MEMCAPS(MEM_CAP_RAM, MEM_CAP_DMA, MEM_CAP_CACHE, MEM_CAP_HP))
define(`PLATFORM_COMP_MEM_CAP', MEMCAPS(MEM_CAP_RAM, MEM_CAP_CACHE))
```

So maybe we can add a new flag e.g. MEM_CAP_DIRECT_ADDRESS if we want to use it to denote if the zone is uncached address used only, but for this kind of zone, do we intend to return cached address/ptr to the user in allocation APIs?

**Comment by keyonjie** [comment]
This is only improvement not mandatory, I have no more time to work on this, closing it.

**Pull request closed without merging**